### PR TITLE
New message dialog

### PR DESCRIPTION
**Pull request by ivolarevic**
# Description

Implemented new behavior of new message dialog:
- when the user scrolls the chat, a simple bottom arrow appears
- when a new message arrives, that arrow expands with the text "You have xy new messages"

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Dev testing.

**Test Configuration**:

* Firmware version: G930FXXS7ETA1
* Hardware: Samsung galaxy s7
* SDK: Android 8, Oreo

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt [hunk 1]
```diff
@@ -623,25 +634,26 @@ class ChatMessagesFragment : BaseFragment(), ChatOnBackPressed {
...
-                is MessagesTimestampFetched -> {
-                    Timber.d("Messages timestamp fetched")
-                    messages = it.messages as MutableList<Message>
-                    //chatAdapter.submitList(it.messages)
-                }
-                is MessageTimestampFetchFail -> Timber.d("Failed to fetch messages timestamp")
-                else -> Timber.d("Other error")
-            }
-        })*/
+//        viewModel.getMessagesTimestampListener.observe(viewLifecycleOwner, EventObserver {
```
Comment (MatkoTom) [inline app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt:646]: I think we can remove this too

File: app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt [hunk 2]
```diff
@@ -927,35 +939,66 @@ class ChatMessagesFragment : BaseFragment(), ChatOnBackPressed {
...
+        if ((scrollYDistance <= 0) && (scrollYDistance > SCROLL_DISTANCE_NEGATIVE)
+            || (scrollYDistance >= 0) && (scrollYDistance < SCROLL_DISTANCE_POSITIVE)
+        ) {
             scrollToPosition()
         } else {
-            // If we received message and keyboard is open:
+            // If we are somewhere up in chat, show new message dialog
+            bindingSetup.cvNewMessages.visibility = View.VISIBLE
+
+            if (newMessagesCount == 1) {
```
Comment (MatkoTom) [inline app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt:962]: Strange to ask the same question in two different places. Maybe extract code in second query to a method and then handle everything in one place.

**Review by MatkoTom — CHANGES_REQUESTED** [review]
Please resolve all comments

File: app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt (hunk 1 above)
Comment (ivolarevic) [reply app/src/main/java/com/clover/studio/exampleapp/ui/main/chat/ChatMessagesFragment.kt:646]: Done

**Review by MatkoTom — APPROVED** [review]
LGTM

**Pull request merged**